### PR TITLE
Add operations for signed arithmetic, and carry condition testing.

### PR DESCRIPTION
**Pull request by robdockins**
For your bikeshedding pleasure, a proposed enhancement to address issue #428.

Still to do: add some sanity checks to the test suite.

### CODE REVIEW

File: docs/CryptolPrims.md [hunk 1]
```diff
@@ -53,6 +71,10 @@ Boolean
     (^)   : a -> a -> a
     (~)   : a -> a
 
+    (==>) : Bit -> Bit -> Bit
+    (/\)  : Bit -> Bit -> Bit
+    (\/)  : Bit -> Bit -> Bit
+
```
Comment (yav) [inline docs/CryptolPrims.md]: These seem unrelated to the signed arithmetic change?

Comment (robdockins) [inline docs/CryptolPrims.md]: It's true, they're a bit unrelated.  The document had fallen a bit out of date in general.

I can split it into separate patches, if you think it's important.

Comment (yav) [inline docs/CryptolPrims.md]: Honestly, it is such a small change that I don't think it is really worth---it is only a change to docs after all. 

I'd say just merge it in.
